subprojects/core/src/integTest/groovy/org/gradle/api/tasks/AbstractIncrementalTasksIntegrationTest.groovy

```diff
@@ -57,7 +57,7 @@ abstract class AbstractIncrementalTasksIntegrationTest extends AbstractIntegrati
     abstract class BaseIncrementalTask extends DefaultTask {
         ${inputDirAnnotation}
         @InputDirectory
-        def File inputDir
+        abstract DirectoryProperty getInputDir()
```

Comment:
I sooo would love to replace `@InputDirectory` with just `@Input`. Wouldn't this be nicer?

```groovy
@Input(incremental = true, pathNormalization = RELATIVE)
abstract DirectoryProperty getInputDir()
```